src/pages/home/ReportScreen.tsx

```diff
@@ -687,8 +687,14 @@ function ReportScreen({route, currentReportID = '', navigation}: ReportScreenPro
         Report.readNewestAction(report?.reportID ?? '');
     }, [report]);
     const mostRecentReportAction = reportActions.at(0);
+    const isMostRecentReportIOU = mostRecentReportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU;
+    const isSingleIOUReportAction = reportActions.filter((action) => action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU).length === 1;
+    const isSingleExpenseReport = ReportUtils.isExpenseReport(report) && isMostRecentReportIOU && isSingleIOUReportAction;
+    const isSingleInvoiceReport = ReportUtils.isInvoiceReport(report) && isMostRecentReportIOU && isSingleIOUReportAction;
     const shouldShowMostRecentReportAction =
         !!mostRecentReportAction &&
```

Comment:
It seems like this case would have already been caught by the `!!mostRecentReportAction` check, could you elaborate why it wasn't?

Reply:
@grgia I'm confused a little, could you please clarify how it would have been caught by the `!!mostRecentReportAction` check?  You can also take a look at my [proposal](https://github.com/Expensify/App/issues/49853#issuecomment-2391367698), I've tried to describe everything there in details.

Reply:
`isMostRecentReportIOU` is `mostRecentReportAction` && is IOU

`isSingleExpenseReport` and `isSingleInvoiceReport` both also include the check `isMostRecentReportIOU`

so I'd think they would have been caught by the `!!mostRecentReportAction`.

Am I misreading the code?

Reply:
Hm, but as you mentioned the  `isMostRecentReportIOU` also includes the check for `IOU` report action type, and that's the main part of this check.